demisto_sdk/commands/unify_packs/packs_unifier.py

```diff
@@ -0,0 +1,191 @@
+    IGNORED_PACKS, ArtifactsManager, ContentObject, create_dirs, delete_dirs,
+    dump_pack, zip_pack_zips, zip_packs)
+
+EX_SUCCESS = 0
+EX_FAIL = 1
+
+
+class PacksUnifier:
+
+    def __init__(self, pack_paths: str, output: str, content_version: str, zip_all: bool, quite_mode: bool = False):
+        default_args = dict(packs=True, zip=True, cpus=os.cpu_count(), suffix='')
+        self.artifacts_manager = PacksManager(
+            pack_paths=pack_paths,
+            artifacts_path=output,
+            content_version=content_version,
+            all_in_one_zip=zip_all,
+            quite_mode=quite_mode,
+            **default_args,
+        )
+
+    def unify_packs(self):
+        """
+        Compress the given packs to one zip file or zip for each pack (depended on the zip_all flag)
+
+        Returns: tuple (the created zip path or the directory of the pack zips, list of the pack names)
+                 or (None, None) if the pack names are empty
+
+        """
+        if self.artifacts_manager.pack_names:
+            self.artifacts_manager.dump_packs()
+            return self.artifacts_manager.output_path, self.artifacts_manager.pack_names
+
+        else:
+            return None, None
+
+
+class PacksManager(ArtifactsManager):
+    def __init__(self, pack_paths: str, all_in_one_zip: bool, quite_mode: bool, **kwargs):
+        super().__init__(**kwargs)
+        self.init_packs(pack_paths)
+        self.zip_all = all_in_one_zip
+        self.quite_mode = quite_mode
+        self.output_path = f'{self.content_uploadable_zips_path}.zip' if self.zip_all \
+            else str(self.content_uploadable_zips_path)
+
+    def init_packs(self, pack_paths):
+        """
+
+        Args:
+            pack_paths: CSV str with the pack paths
+
+        init dict that map pack name to Pack object and init also the pack_names property
+
+        """
+        self.packs = {}
+        for path_str in arg_to_list(pack_paths):
+            path = Path(path_str)
+            if len(path.parts) == 2 and path.parts[0] == PACKS_DIR:  # relative path from Packs/...
+                path = self.content.path / path
+
+            if not os.path.exists(path):
+                click.secho(f'Error: Given input path: {path} does not exist, ignored', fg='bright_red')
+                continue
+
+            self.packs.update({path.name: Pack(path)})
+
+        self.pack_names = [*self.packs]
+
+    def get_relative_pack_path(self, content_object: ContentObject):
+        """
+
+        Args:
+            content_object: the object to get the relative path for
+
+        Returns:
+            the path of the given object relative from the pack directory, for example HelloWorld/Scripts
+
+
+        """
+        for part in content_object.path.parts:
+            if part in self.packs:
+                return content_object.path.relative_to(self.packs[part].path.parent)
+
+    def get_base_path(self):
```

Comment:
Just to make sure, you need this method in order to override the `super` class method, right?

Reply:
Exactly